lib/matplotlib/constrained_layout.py

```diff
@@ -0,0 +1,696 @@
+"""
```

Comment:
Does this module need to be a public API?  I know we have had a tendency to expose most of our internals a public submodules but perhaps it's a good time to start making them private...